orttraining/orttraining/test/python/orttraining_test_ortmodule_fallback.py

```diff
@@ -558,3 +562,49 @@ def test_ortmodule_fallback_warn_message(is_training, persist_fallback):
             with pytest.warns(UserWarning) as warning_record:
                 ort_model(inputs)
         assert "Fallback to PyTorch due to exception" in str(warning_record[0].message.args[0])
+
+    del os.environ['ORTMODULE_SKIPCHECK_POLICY']
+
+
+@pytest.mark.parametrize("is_training,fallback_enabled,matching_policy,persist_fallback",
+                         list(itertools.product([True, False], repeat=4)))
+def test_ortmodule_fallback_with_skipcheck_reset(is_training, fallback_enabled, matching_policy, persist_fallback):
+    # is_training: True for torch.nn.Module training model, eval mode otherwise
+    # fallback_enabled: True PyTorch executes the forward graph instead of ORT backend
+    # matching_policy: True matches FALLBACK_UNSUPPORTED_DATA policy to ORTModuleDeviceException exception.
+    #   Otherwise, an incorrect policy (FALLBACK_UNSUPPORTED_DEVICE) is used to verify that the fallback does not happen
+
+    if fallback_enabled:
+        if matching_policy:
+            policy = 'FALLBACK_UNSUPPORTED_DATA'
+        else:
+            policy = 'FALLBACK_UNSUPPORTED_DEVICE'
+    else:
+        policy = 'FALLBACK_DISABLE'
+    os.environ['ORTMODULE_FALLBACK_POLICY'] = policy
+    os.environ['ORTMODULE_FALLBACK_RETRY'] = str(not persist_fallback)
+
+    pt_model = MyStrNet()
+    ort_model = ORTModule(copy.deepcopy(pt_model))
+    inputs = torch.randn(1, 2)
+
+    ort_model.train(is_training)
+    pt_model.train(is_training)
+
+    assert ort_model._torch_module._execution_manager(is_training)._skip_check == \
+        (_utils._SkipCheck.SKIP_CHECK_DEVICE | _utils._SkipCheck.SKIP_CHECK_BUILD_GRADIENT | 
+        _utils._SkipCheck.SKIP_CHECK_EXECUTION_AGENT)
+
+    for i in range(3):
+        if fallback_enabled:
+            if matching_policy:
+                ort_out = ort_model(inputs, 'hello')
+                assert ort_model._torch_module._execution_manager(is_training)._skip_check == _utils._SkipCheck.SKIP_CHECK_DISABLED
```

Comment:
I think I am missing something.
What is the point of having the ORT skip check flag disabled if the model is still executed by pytorch? With skip check or not, pytorch is running the code, so the flag doesnt make any difference in terms of perm or execution flow